pyro/primitives.py

```diff
@@ -248,7 +249,7 @@ def __enter__(self):
         self.dim = _DIM_ALLOCATOR.allocate(self.name, self.dim)
         if self._wrapped:
             try:
-                self._scale_messenger = poutine.scale(scale=float(self.size / self.subsample_size))
+                self._scale_messenger = poutine.scale(scale=self.size / self.subsample_size)
```

Comment:
I believe this needs to be a float. I've seen jit errors due to the jit not respecting `from future import __division__` and therefore doing integer arithmetic here. Have the changes in this PR fixed those errors?

Reply:
Good point. I suppose `float(..)` won't help in that case, since it will just cast the end result after the integer arithmetic into a python float. But what we can do is check that if size is a tensor then convert it to a float tensor during the trace phase.

Reply:
How about
```py
def torch_float(x):
    return x.float() if isinstance(x, torch.Tensor) else float(x)

poutine.scale(scale=torch_float(self.size) / torch_float(self.subsample_size))
```

Reply:
Great, will add this to util.